openedx/core/djangoapps/programs/utils.py

```diff
@@ -463,26 +463,44 @@ def _collect_one_click_purchase_eligibility_data(self):
         if is_learner_eligible_for_one_click_purchase:
             for course in self.data['courses']:
                 add_course_sku = False
+                unpublished_enrollment = False
+                unpublished_course_runs = filter(lambda run: run['status'] == 'unpublished', course['course_runs'])
                 published_course_runs = filter(lambda run: run['status'] == 'published', course['course_runs'])
                 if len(published_course_runs) == 1:
                     # Look at the course runs for a course and determine if the course SKU should be added.
-                    course_run = published_course_runs[0]
-                    (enrollment_mode, active) = CourseEnrollment.enrollment_mode_for_user(
-                        self.user,
-                        CourseKey.from_string(course_run['key'])
-                    )
-
-                    if enrollment_mode is not None and active is not None:
-                        # Check all the applicable seat types
-                        # this will also check for no-id-professional as professional
-                        applicable_seat = any(seat_type in enrollment_mode for seat_type in applicable_seat_types)
-
-                        # If no applicable seat is found add the course SKU to the list
-                        if not applicable_seat or not active:
+
```

Comment:
Seems like a lot of this is repeated. A lot of it could be cleaned up with some nice helper functions